peps/pep-0812.rst

```diff
@@ -0,0 +1,651 @@
+``complex(-1, 0.0)``).  Though, there is again no difference for equality.
+
+Here we list variants of backward incompatible behavior:
+
+.. code:: pycon
+
+   >>> type(1j)  # was "<class 'complex'>"
+   <class 'imaginary'>
+   >>> type(-123j)  # was "<class 'complex'>"
+   <class 'imaginary'>
+   >>> -123j  # was "(-0-123j)"
+   -123j
+   >>> complex(+0.0, 1)  # was "1j"
+   (0.0+1j)
+   >>> complex(-0.0, 1)  # was "(-0+1j)"
+   (-0.0+1j)
+   >>> complex('1j')  # was "1j"
+   (0.0+1j)
+   >>> format(1j, "f")  # was '0.000000+1.000000j'
+   '1.000000j'
+   >>> format(-1j, "f")  # was '-0.000000-1.000000j'
+   '-1.000000j'
+   >>> +0.0+1j  # was "1j"
+   (0.0+1j)
+   >>> -0.0+1j  # was "1j"
+   (-0.0+1j)
+   >>> float('inf')*1j  # was "(nan+infj)"
+   infj
+   >>> float('nan')*1j  # was "(nan+nanj)"
+   nanj
+   >>> -0.0*1j  # was "(-0-0j)"
+   -0j
+
+
+Working with the initial implementation shows, that most test failures in the
+CPython test suite come from cases, where imaginary literals are used just as
+"some complex numbers", to produce exceptions.  Running the `mpmath
+<https://github.com/mpmath/mpmath>`_ (develompment version) test suite shows
+only two test failures.  The `NumPy <https://github.com/numpy/numpy>`_ (v2.3.4)
+has ~16 broken tests.
+
+
+How to Teach This
+=================
+
+While internaly complex arithmetic will be more complicated (but not too much,
+see `Reference Implementation <Reference Implementation_>`_), its semantics
+will be more close to usual mathematical notation in textbooks on complex
+analysis, much less place for confusion of newcomers.  Roughly speaking, it
+will be the floating-point arithmetic, augmented by the special algebraical
+symbol ``1j``, which square is ``-1.0``.
+
+
+Reference Implementation
+========================
+
+A draft implementation is available in a
+https://github.com/skirpichev/cpython/pull/1
+
+
+Rejected Ideas
+==============
+
+We might try to implement complex arithmetic, that will treat specially ---
+just like instances of imaginary type proposed --- complex numbers of the form
+``complex(0.0, y)``.  But such proposal alter arithmetic rules on the set of
+complex numbers itself, in particular ``complex(a, b) + complex(c, d)`` will
+not be exactly equal to ``complex(a + c, b + d)`` anymore.
+
+The set of imaginary numbers with a special treatment in complex arithmetic
+might implemented differently, as a distinct form for the complex type
+constructor, say ``complex(imag=y, pure=True)``.  However, experiments show
+that such implementation is more complicated internally and more hard to
+explain then a dedicated concept of imaginary numbers as a subtype of complex.
+
+
+Open Issues
+===========
+
+The PEP doesn't expose new subtype as a builtin, say ``imaginary``, but maybe
+we should?  This looks redundant, as all imaginary values could be obtained by
+scaling imaginary unit, i.e. ``imaginary(x) == float(x)*1j``.
```

Comment:
```suggestion
The PEP doesn't expose a new subtype as a builtin, say ``imaginary``, but maybe
we should?  This looks redundant, as all imaginary values could be obtained by
scaling imaginary units, i.e. ``imaginary(x) == float(x)*1j``.
```

Reply:
This changes meaning.  "the imaginary unit" - will be correct, I think.